chapter01/1.3 - URLify/urlify.js

```diff
@@ -0,0 +1,24 @@
+var urlify = function(str, length) {
+  // have a pointer to check from start to end
+  var strArr = str.split('');
+  var pointer = 0;
+  while (pointer < str.length) {
+    if (strArr[pointer] === ' ') {
+      // *** needs more work here, a little wierd
+      // not handling trailing spaces properly
+      for (var i = str.length - 1; i > pointer + 3; i--) {
+        strArr[i] = str[i - 2];
+      }
+      strArr[pointer] = '%';
+      strArr[pointer+1] = '2';
```

Comment:
👍 Great points, I like this code. I'll look more into the other files later today 👍